### PR TITLE
send RealPhone verification message with iQ

### PR DESCRIPTION
**Pull request by groovecoder**
Add IQ_FOR_VERIFICATION env var and setting.

Refactor api.phones.views._send_iq_sms into PhonesConfig.send_iq_sms

Use PhonesConfig.send_iq_sms to send real phone verification when IQ_FOR_VERIFICATION is True.

## How to test:
1. [Set up local phone dev with iQ](https://github.com/mozilla/fx-private-relay/blob/ec1a69e1432248c98e9105f71ed8c1277695ddee/docs/end-to-end-local-phone-dev-iq.md).
   * Note: this PR updates that doc
   * [x] When you get to the step "Go to http://127.0.0.1:8000/phone/ and verify your real phone: it will send an SMS via inteliquent from the IQ_MAIN_NUMBER above." - verify it sends the verification message from the `IQ_MAIN_NUMBER`!

# Screenshot (if applicable)
image

# Checklist (Definition of Done)
- [x] Product Owner [accepted the User Story](https://mozilla-hub.atlassian.net/browse/MPP-2850?focusedCommentId=667856) (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] I've added or updated relevant docs in the docs/ directory
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to ~stage~ dev.
- [x] ~All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/frontend/src/styles/tokens.scss`).~
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] ~l10n changes have been submitted to the l10n repository, if any.~

### CODE REVIEW

**Comment by jwhitlock** [comment]
The integrated test worked on my dev machine. After adding:

```
IQ_FOR_VERIFICATION=True
IQ_MAIN_NUMBER="+1..."
```

and restarting, signing up sent a text from `IQ_MAIN_NUMBER`. Now to look at the code and tests...

File: api/tests/iq_views_tests.py [hunk 1]
```diff
@@ -54,6 +52,13 @@ def mocked_twilio_client():
         yield mock_twilio_client
 
 
+def _make_real_phone_with_mock_iq(phone_user, **kwargs):
```
Comment (jwhitlock) [inline api/tests/iq_views_tests.py:55]: 👍 nice helper function, cleans up the test setup

File: frontend/src/components/dashboard/aliases/BlockLevelSlider.tsx [hunk 2]
```diff
@@ -465,7 +465,7 @@ function getLabelForBlockLevel(
     case "none":
       return l10n.getString("profile-promo-email-blocking-option-none");
     case "promotional":
-      return l10n.getString("profile-promo-email-blocking-option-promotionals");
+      return l10n.getString("profile-promo-email-blocking-option-promotions");
```
Comment (jwhitlock) [inline frontend/src/components/dashboard/aliases/BlockLevelSlider.tsx:468]: Is this change intended in this PR?

File: phones/apps.py [hunk 3]
```diff
@@ -44,3 +48,19 @@ def twilio_validator(self) -> RequestValidator:
         if not settings.TWILIO_AUTH_TOKEN:
             raise Exception("Must define TWILIO_AUTH_TOKEN")
         return RequestValidator(settings.TWILIO_AUTH_TOKEN)
+
+    def send_iq_sms(self, to_num: str, from_num: str, text: str) -> None:
```
Comment (jwhitlock) [inline phones/apps.py:52]: Why is this a method on the AppConfig class? It seems like it would belong in `iq_utils.py` or similar.

File: phones/models.py [hunk 4]
```diff
@@ -124,6 +133,10 @@ def get_last_text_sender(relay_number: "RelayNumber") -> Optional["InboundContac
     return latest
 
 
+def iq_fmt(e164_number: str) -> str:
```
Comment (jwhitlock) [inline phones/models.py:136]: 👍 this is a good place for this to live

File: privaterelay/settings.py [hunk 5]
```diff
@@ -201,9 +201,15 @@
 DAYS_PER_BILLING_CYCLE = config("DAYS_PER_BILLING_CYCLE", 30, cast=int)
 MAX_DAYS_IN_MONTH = 31
 IQ_ENABLED = config("IQ_ENABLED", False, cast=bool)
+IQ_FOR_VERIFICATION = config("IQ_FOR_VERIFICATION", False, cast=bool)
 IQ_FOR_NEW_NUMBERS = config("IQ_FOR_NEW_NUMBERS", False, cast=bool)
+IQ_MAIN_NUMBER = config("IQ_MAIN_NUMBER", "")
 IQ_OUTBOUND_API_KEY = config("IQ_OUTBOUND_API_KEY", None)
 IQ_INBOUND_API_KEY = config("IQ_INBOUND_API_KEY", "")
+IQ_MESSAGE_PATH = config(
```
Comment (jwhitlock) [inline privaterelay/settings.py:209]: When would we customize `IQ_MESSAGE_PATH`? Putting it in settings suggests to me that it may change in different deployments, like stage vs production

File: requirements.txt [hunk 6]
```diff
@@ -28,11 +28,6 @@ requests==2.28.2
 sentry-sdk==1.16.0
 whitenoise==6.4.0
 
-# issue #3243
```
Comment (jwhitlock) [inline requirements.txt:31]: Ah, I know this one - maybe a rebase on main will make some of these other commits go away.

**Review by jwhitlock — APPROVED** [review]
Thanks @groovecoder, this code works for me, and the test updates are an improvement. I've got a few questions, including some weird changes that may go away with a rebase. No blockers though.

File: phones/apps.py (hunk 3 above)
Comment (groovecoder) [reply phones/apps.py:52]: I'll move it to an `iq_utils.py` module. I just hate when I find myself making a `utils` module - I feel like I've probably failed in my architectural design or thinking somehow. But it makes sense.

File: privaterelay/settings.py (hunk 5 above)
Comment (groovecoder) [reply privaterelay/settings.py:209]: I found myself duplicating this code in a few places (models, api view, tests):
```python
iq_publish_message_url = f"{IQ_MESSAGE_API_ORIGIN}/msgbroker/rest/publishMessages"
```

So, I wanted to move it "up" into a module that is easy to import in all those places.

But that's weird ... in my local branch I have:
```python
  IQ_MESSAGE_API_ORIGIN = config(
      "IQ_MESSAGE_API_ROOT", "https://messagebroker.inteliquent.com"
  )
  IQ_MESSAGE_PATH = "/msgbroker/rest/publishMessages"
  IQ_PUBLISH_MESSAGE_URL = f"{IQ_MESSAGE_API_ORIGIN}{IQ_MESSAGE_PATH}"
```

I.e., I don't pull the `IQ_MESSAGE_PATH` from an env var - I just hard-code it. If this looks okay, maybe I just need to figure out why this PR seems to have some changes I didn't mean for it to have, and it's misisng some changes that it should have.

**Comment by groovecoder** [comment]
Hrm. I rebased on latest main but it said everything was up to date. I'll try to dig into what happened here. 😢

**Comment by groovecoder** [comment]
Superseding with https://github.com/mozilla/fx-private-relay/pull/3269 which has cherry-picked commits and review changes.
